client/landing/stepper/declarative-flow/internals/components/step-route/test/index.tsx

```diff
@@ -0,0 +1,175 @@
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import { renderWithProvider } from 'calypso/test-helpers/testing-library';
+import StepRoute from '../';
+import { NavigationControls } from '../../../types';
+
+jest.mock( 'calypso/signup/storageUtils' );
+jest.mock( 'calypso/state/current-user/selectors' );
+jest.mock( 'calypso/landing/stepper/declarative-flow/internals/analytics/record-step-start' );
+jest.mock( 'calypso/landing/stepper/hooks/use-intent' );
+jest.mock( 'calypso/landing/stepper/hooks/use-selected-design' );
+jest.mock( 'calypso/lib/analytics/page-view' );
+
+const RegularStep: StepperStep = {
+	slug: 'some-step-slug',
+	asyncComponent: jest.fn(), // For the purpose of this test, we don't need to render the using lazy strategy
+};
+
+const RequiresLoginStep: StepperStep = {
+	...RegularStep,
+	requiresLoggedInUser: true,
+};
+
+const FakeStep: FC< StepProps > = () => {
+	return <div>Step Content</div>;
+};
+
+const FakeFlow: Flow = {
+	name: 'some-flow',
+	useSteps: () => [ RegularStep ],
+	useStepNavigation: () => ( {
+		submit: () => {},
+	} ),
+
+	isSignupFlow: false,
+};
+
+const fakeRenderStep = ( step: AsyncStepperStep ) => {
+	const StepComponent = FakeStep;
+	const navigationControls = {} as NavigationControls;
+	const stepData = {} as any;
+
+	return (
+		<StepComponent
+			navigation={ navigationControls }
+			flow={ FakeFlow.name }
+			variantSlug={ FakeFlow.variantSlug }
+			stepName={ step.slug }
+			data={ stepData }
+		/>
+	);
+};
+const render = ( { step, renderStep = fakeRenderStep } ) => {
+	return renderWithProvider(
+		<MemoryRouter>
+			<Suspense fallback={ null }>
+				<StepRoute
+					step={ step }
+					flow={ FakeFlow }
+					showWooLogo={ false }
+					renderStep={ renderStep }
+				/>
+			</Suspense>
+		</MemoryRouter>
+	);
+};
+describe( 'StepRoute', () => {
+	// we need to save the original object for later to not affect tests from other files
+	const originalLocation = window.location;
+
+	beforeAll( () => {
+		Object.defineProperty( window, 'location', {
+			value: { ...originalLocation, assign: jest.fn() },
+		} );
+		( useIntent as jest.Mock ).mockReturnValue( 'build' );
+		( useSelectedDesign as jest.Mock ).mockReturnValue( { design_type: 'premium' } );
+	} );
+
+	beforeEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	afterAll( () => {
+		Object.defineProperty( window, 'location', originalLocation );
+	} );
+
+	it( 'renders the step content', async () => {
+		const { getByText } = render( { step: RegularStep } );
+
+		await waitFor( () => expect( getByText( 'Step Content' ) ).toBeInTheDocument() );
+	} );
+
+	describe( 'requiresLoggedInUser', () => {
+		it( 'redirects users to the login page if the step requires a logged in user', async () => {
+			render( { step: RequiresLoginStep } );
+
+			await waitFor( () =>
+				expect( window.location.assign ).toHaveBeenCalledWith(
+					'/start/account/user-social?variationName=some-flow&redirect_to=%2F&toStepper=true'
```

Comment:
We should probably update this to include the step name.

Reply:
This feels like it's an artifact of the way the rendering is done. Might it make sense to parameterise either `basename` or `initialEntries` in the `MemoryRouter` so we can clearly control/dictate what the current location will show up as?

Or should we push that logic into separate tests for `useLoginUrlForFlow()`?

Reply:
Nah, this comment is irrelevant now that you clarified that we're just using the current URL to bring the user back to.